libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/AnalyticsPublisherService.java

```diff
@@ -59,11 +52,11 @@ public AnalyticsPublisherService() {
     public static void startActionPublish(Context context) {
         final Intent intent = new Intent(context, AnalyticsPublisherService.class);
         intent.setAction(ACTION_PUBLISH);
-        context.startService(intent);
+        enqueueWork(context, AnalyticsPublisherService.class, JOB_ID, intent);
```

Comment:
`enqueueWork` does the heavy lifting of whether to invoke `startService()` on older devices or use `JobScheduler` on newer devices.